src/toolkits/activity_classification/ac_data_iterator.cpp

```diff
@@ -372,14 +517,15 @@ simple_data_iterator::preprocessed_data simple_data_iterator::preprocess_data(
   return result;
 }
 
-simple_data_iterator::simple_data_iterator(const parameters& params)
-  : data_(preprocess_data(params)),
-    num_samples_per_prediction_(params.prediction_window),
-    num_predictions_per_chunk_(params.predictions_in_chunk),
-    range_iterator_(data_.chunks.range_iterator()),
-    next_row_(range_iterator_.begin()),
-    end_of_rows_(range_iterator_.end())
-{}
+size_t sample = 0;
+
+simple_data_iterator::simple_data_iterator(const parameters &params)
+    : data_(preprocess_data(params)),
+      num_samples_per_prediction_(params.prediction_window),
+      num_predictions_per_chunk_(params.predictions_in_chunk),
+      range_iterator_(data_.chunks.range_iterator()),
+      next_row_(range_iterator_.begin()), end_of_rows_(range_iterator_.end()),
+      sample_in_row_(0), training(params.verbose) {}
```

Comment:
It seems strange to me that you're initializing `training` to the values of `params.verbose`.

Reply:
I renamed the parameter to ```is_train```. It does both but it makes sense to called ```is_train``` rather than ```verbose```.